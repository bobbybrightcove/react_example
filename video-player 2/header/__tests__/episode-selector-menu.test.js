import React from 'react';
import renderer from 'react-test-renderer';
import EpisodeSelectorMenu from '../episode-selector-menu';
import contentItem from '../../__tests__/content-item.json';

describe('EpisodeSelectorMenu', () => {
  const player = {};
  const onSelectEpisodeSelectorOption = jest.fn();
  const setSeasonSelectorVisibility = jest.fn();
  const seasonNumber = 1;
  const episode = contentItem;

  const selected = {
    id: 'c4db2204-16f2-469f-8bc1-5c2244295930',
    title: 'Deep Water',
    description: 'Jih iza buh bihewsec go ihaojo dimdu jaf wadu tap abmoado fajropab. Era zuzjahok tonegin idba lid jizmedtis cevi weraj wettuw op ban uhmowlis gavaeli ebe zoko epu. Vub ciuj tuhjem ma dup elwuefi saodil agumur kahpef fezvifkoj uburoruj toci ijoneza wuldafose. Niporna dop zuv rewbak ejifitem du owpeloz we rusam nabse purlol hehawruk adabaga ba.',
  };

  it('should render with no selection', () => {
    const component = renderer.create(
      <EpisodeSelectorMenu
        options={episode.series.seasons[seasonNumber].episodes}
        player={player}
        selected={{}}
        onSelectEpisodeSelectorOption={onSelectEpisodeSelectorOption}
        setSeasonSelectorVisibility={setSeasonSelectorVisibility}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with episode selected', () => {
    const component = renderer.create(
      <EpisodeSelectorMenu
        options={episode.series.seasons[seasonNumber].episodes}
        player={player}
        selected={selected}
        onSelectEpisodeSelectorOption={onSelectEpisodeSelectorOption}
        setSeasonSelectorVisibility={setSeasonSelectorVisibility}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
