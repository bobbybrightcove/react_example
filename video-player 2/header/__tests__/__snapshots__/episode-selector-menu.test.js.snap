// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeSelectorMenu should render with episode selected 1`] = `
<div>
  <p>
    <img
      alt="Back"
      onClick={[Function]}
      src="back.svg"
    />
    <span>
      Season 
      2
    </span>
  </p>
  <div
    className="rc-collapse episodeSelectorOptions"
    style={undefined}
  >
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        A Lie Agreed Upon, Part 1
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        A Lie Agreed Upon, Part 2
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        New Money
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Requiem for a Gleet
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Complications
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Something Very Expensive
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        E.B. Was Left Out
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Childish Things
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Amalgamation and Capital
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Advances, None Miraculous
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        The Whores Can Come
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Boy the Earth Talks To
      </div>
    </div>
  </div>
</div>
`;

exports[`EpisodeSelectorMenu should render with no selection 1`] = `
<div>
  <p>
    <img
      alt="Back"
      onClick={[Function]}
      src="back.svg"
    />
    <span>
      Season 
      2
    </span>
  </p>
  <div
    className="rc-collapse episodeSelectorOptions"
    style={undefined}
  >
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        A Lie Agreed Upon, Part 1
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        A Lie Agreed Upon, Part 2
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        New Money
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Requiem for a Gleet
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Complications
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Something Very Expensive
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        E.B. Was Left Out
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Childish Things
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Amalgamation and Capital
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Advances, None Miraculous
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        The Whores Can Come
      </div>
    </div>
    <div
      className="rc-collapse-item episodeSelectorOption"
      id={undefined}
      style={undefined}
    >
      <div
        aria-expanded={false}
        className="rc-collapse-header episodeSelectorOptionName"
        onClick={[Function]}
        role="tab"
      >
        <i
          className="arrow"
        />
        Boy the Earth Talks To
      </div>
    </div>
  </div>
</div>
`;
